### PR TITLE
Add close method, remove mu member

### PR DESCRIPTION
**Pull request by alexflint**
Based on #2

Added test for real simultaneous lock attempt.

Removed debug Printf from test script.

### CODE REVIEW

File: filemutex_flock.go [hunk 1]
```diff
@@ -60,6 +53,15 @@ func (m *FileMutex) RUnlock() error {
 	if err := syscall.Flock(m.fd, syscall.LOCK_UN); err != nil {
 		return err
 	}
-	m.mu.RUnlock()
 	return nil
 }
+
+// Close does an Unlock() combined with closing and unlinking the associated
```
Comment (rosenhouse) [inline filemutex_flock.go]: Comment still says "unlinking".

File: filemutex_test.go [hunk 2]
```diff
@@ -40,7 +41,20 @@ func TestRLockUnlock(t *testing.T) {
...
+	dir, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	path := filepath.Join(dir, "x")
+	m, err := New(path)
+	require.NoError(t, err)
+
+	m.Lock()
+	m.Close()
```
Comment (rosenhouse) [inline filemutex_test.go]: may as well `require.NoError` on the returns from these two lines

File: filemutex_test.go [hunk 3]
```diff
@@ -192,3 +206,109 @@ func TestRUnlockErrorsAreRecoverable(t *testing.T) {
...
+			state = "waiting on lock" // for m to unlock before m2 can lock
+			ch <- struct{}{}
+		}()
+		m2.Lock()
+		state = "acquired"
+		ch <- struct{}{}
+
+		<-ch
+		m2.Unlock()
+		state = "closed"
```
Comment (rakelkar) [inline filemutex_test.go]: nit: should be released? (its not closed)

**Review by rakelkar — APPROVED** [review]
LGTM barring nits

**Comment by rakelkar** [comment]
hey @alexflint possible to get this in?
Tests pass on windows & linux...

C:\Users\rakelkar\go\src\github.com\alexflint\go-filemutex (add-close)
λ go test -v ./...
=== RUN   TestLockUnlock
--- PASS: TestLockUnlock (0.01s)
=== RUN   TestRLockUnlock
--- PASS: TestRLockUnlock (0.00s)
=== RUN   TestClose
--- PASS: TestClose (0.00s)
=== RUN   TestSequentialLock
--- PASS: TestSequentialLock (0.00s)
=== RUN   TestLockErrorsAreRecoverable
--- PASS: TestLockErrorsAreRecoverable (0.00s)
=== RUN   TestUnlockErrorsAreRecoverable
--- PASS: TestUnlockErrorsAreRecoverable (0.00s)
=== RUN   TestRLockErrorsAreRecoverable
--- PASS: TestRLockErrorsAreRecoverable (0.00s)
=== RUN   TestRUnlockErrorsAreRecoverable
--- PASS: TestRUnlockErrorsAreRecoverable (0.00s)
=== RUN   TestSimultaneousLock
--- PASS: TestSimultaneousLock (0.05s)
=== RUN   TestSimultaneousLockWithClose
--- PASS: TestSimultaneousLockWithClose (0.06s)
PASS
ok      github.com/alexflint/go-filemutex       0.303s

**Comment by alexflint** [comment]
FYI several tests were failing with data races so I've removed them from this pr and will add them later.

**Pull request merged**